### PR TITLE
Made accessor_properties a ONEAPI extension and added the buffer_location extension

### PR DESCRIPTION
**Pull request by GarveyJoe**
As part of this change, the new functionality was extracted from property_list into accessor_property_list in the ONEAPI namespace.  Another significant change was that the class that represents an instance of a property is now implementation-defined and only accessed through a global variable.

### CODE REVIEW

**Review by jbrodman — APPROVED** [review]

File: sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+== Modifications to the SYCL 2020 Pre-Provisional Specification
+
+=== Section 4.7.6.9.2 Device buffer accessor properties
+
+Add a new property and global variable to the listing:
+
+```c++
+namespace sycl {
+...
+namespace INTEL {
```
Comment (MrSidims) [inline sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc:60]: @GarveyJoe have you considered swapping `INTEL` and `property` namespaces in the nesting?
The thing I find confusing in the namespace nesting, is that property namespace is put inside of INTEL (aka hardware extension) namespace, but we will have some other properties, that will have not only FPGA usage, hence these properties shall be placed in ONEAPI (aka language extensions) namespace.

Comment (GarveyJoe) [reply sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc:60]: I was trying to be consistent with the spirit of the rules from the SYCL 2020 provisional spec which indicates that vendor extensions should be in their own namespace nested in the sycl namespace.  @gmlueck is this the recommended way to nest these namespaces?

Comment (MrSidims) [reply sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc:60]: That makes sense, Thanks.

Comment (gmlueck) [reply sycl/doc/extensions/accessor_properties/SYCL_INTEL_buffer_location.asciidoc:60]: If you are asking whether this extension should be in the `INTEL` vs `ONEAPI` namespace, I think @jbrodman is the right person to answer.

Regarding the order of the namespaces `sycl::INTEL::property` vs. `sycl::property::INTEL`, I think the first is preferable (which is what you have).

**Pull request merged**